rcsb/utils/dictionary/DictMethodEntityInstanceHelper.py

```diff
@@ -2034,40 +2037,90 @@ def buildEntityInstanceAnnotations(self, dataContainer, catName, **kwargs):
+                    logger.info("Annotations  - entryId %s asymId %s rTupTmpL: %r", entryId, asymId, rTupTmpL)
+                    rTupS = set(rTupTmpL)
+                    logger.info("Annotations Set - entryId %s asymId %s rTupS: %r", entryId, asymId, rTupS)
+                    for rTup in rTupS:
+                        targetCompId, connectType, partnerEntityType, partnerAsymId = rTup[0], rTup[1], rTup[2], rTup[3]
+                        if connectType in ["covalent bond"]:
+                            aL.append(
+                                {
+                                    "type": "HAS_COVALENT_LINKAGE",
+                                    "annotation_id": f"COVALENT_LINKAGE_{jj}",
+                                    "entity_id": asymIdD[asymId],
+                                    "asym_id": asymId,
+                                    "auth_asym_id": authAsymId,
+                                    "comp_id": targetCompId,
+                                    "provenance_source": "Primary Data",
+                                    "assignment_version": "V1.0",
+                                    "description": f"{targetCompId} has {connectType} with {partnerEntityType} instance {partnerAsymId} in model 1",
+                                }
+                            )
+                        elif connectType in ["metal coordination"]:
+                            aL.extend([
+                                {
+                                    "type": "HAS_METAL_COORDINATION_LINKAGE",
+                                    "annotation_id": f"METAL_COORDINATION_LINKAGE_{jj}",
+                                    "entity_id": asymIdD[asymId],
+                                    "asym_id": asymId,
+                                    "auth_asym_id": authAsymId,
+                                    "comp_id": targetCompId,
+                                    "provenance_source": "Primary Data",
+                                    "assignment_version": "V1.0",
+                                    "description": f"{targetCompId} has {connectType} with {partnerEntityType} instance {partnerAsymId} in model 1",
+                                },
+                                {
+                                    "type": "HAS_NO_COVALENT_LINKAGE",
+                                    "entity_id": asymIdD[asymId],
+                                    "asym_id": asymId,
+                                    "auth_asym_id": authAsymId,
+                                    "comp_id": targetCompId,
+                                    "provenance_source": "Primary Data"
+                                }
+                            ])
+                        else:
+                            continue
+                        #
+                        jj += 1
+
+                if len(aL) == 0:
+                    aL.append({
+                        "type": "HAS_NO_COVALENT_LINKAGE",
```

Comment:
Thanks @piehld. Is it better to handle the logic of obtaining annotation `type` in `DictMethodCommonUtils` and only deal with the assignments here in `DictMethodEntityInstanceHelper` keeping with the usual practice?

Reply:
@brindakv The reason I didn't make these changes in the CommonUtils is because the method that's called in that file is `getBoundNonpolymersByInstance`, which as its name implies is getting only the bound non-polymers. And that method is called elsewhere in the code too, so making a change in that method would affect other calls of it.

Also, the way that method works is by iterating over the `struct_conn` items, which only represent bound instances. So, it wouldn't have the scope of being able to iterate over all other unbound instances...or at least that's what I could determine...

Am I understanding your question correctly?

Reply:
Added a note about this in the PR description.